server/shared/configFetcher.go

```diff
@@ -54,25 +55,30 @@ func (cf *configFetcher) Fetch() (files []*remote.FileMeta, err error) {
 				// test .woodpecker/ folder
 				// if folder is not supported we will get a "Not implemented" error and continue
 				files, err = cf.remote_.Dir(cf.user, cf.repo, cf.build, ".woodpecker")
-				if err == nil {
-					return filterPipelineFiles(files), nil
+				files = filterPipelineFiles(files)
+				if err == nil && len(files) != 0 {
+					return files, nil
 				}
 
 				file, err = cf.remote_.File(cf.user, cf.repo, cf.build, ".woodpecker.yml")
-				if err == nil {
+				if err == nil && len(file) != 0 {
 					return []*remote.FileMeta{{
 						Name: ".woodpecker.yml",
 						Data: file,
 					}}, nil
 				}
 
 				file, err = cf.remote_.File(cf.user, cf.repo, cf.build, ".drone.yml")
-				if err == nil {
+				if err == nil && len(file) != 0 {
 					return []*remote.FileMeta{{
 						Name: ".drone.yml",
 						Data: file,
 					}}, nil
 				}
+
+				if err == nil && len(files) == 0 {
+					return nil, fmt.Errorf("ConfigFetcher: Fallback did not found config")
+				}
 			}
 
 			return nil, err
```

Comment:
I think this line is stopping the retry loop. Should we just remove the retry loop?

Reply:
can we leave it for now?

I like to add logging (more precice one!) - if we do have that we can drop it

Reply:
We can just leave it as it is now and add a TODO, saying the retry loop is inactive and could maybe be deleted in future.